gadget/install/install.go

```diff
@@ -167,53 +166,7 @@ func Run(gadgetRoot, device string, options Options, observer gadget.ContentObse
 		return fmt.Errorf("cannot store recovery key: %v", err)
 	}
 
-	// TODO:UC20: binaries are EFI/bootloader-specific, hardcoded for now
-	loadChain := []bootloader.BootFile{
-		// the path to the shim EFI binary
-		bootloader.NewBootFile("", filepath.Join(boot.InitramfsUbuntuSeedDir, "EFI/boot/bootx64.efi"), bootloader.RoleRecovery),
-		// the path to the recovery grub EFI binary
-		bootloader.NewBootFile("", filepath.Join(boot.InitramfsUbuntuSeedDir, "EFI/boot/grubx64.efi"), bootloader.RoleRecovery),
-		// the path to the run mode grub EFI binary
-		bootloader.NewBootFile("", filepath.Join(boot.InitramfsUbuntuBootDir, "EFI/boot/grubx64.efi"), bootloader.RoleRunMode),
-	}
-	if options.KernelPath != "" {
-		// the path to the kernel EFI binary
-		loadChain = append(loadChain, bootloader.NewBootFile("", options.KernelPath, bootloader.RoleRunMode))
-	}
-
-	// Get the expected kernel command line for the system that is currently being installed
-	cmdline, err := boot.ComposeCandidateCommandLine(options.Model)
-	if err != nil {
-		return fmt.Errorf("cannot obtain kernel command line: %v", err)
-	}
-
-	// Get the expected kernel command line of the recovery system we're installing from
-	recoveryCmdline, err := boot.ComposeRecoveryCommandLine(options.Model, options.SystemLabel)
-	if err != nil {
-		return fmt.Errorf("cannot obtain recovery kernel command line: %v", err)
-	}
-
-	kernelCmdlines := []string{
-		cmdline,
-		recoveryCmdline,
-	}
-
-	sealKeyParams := secboot.SealKeyParams{
-		ModelParams: []*secboot.SealKeyModelParams{
-			{
-				Model:          options.Model,
-				KernelCmdlines: kernelCmdlines,
-				EFILoadChains:  [][]bootloader.BootFile{loadChain},
-			},
-		},
-		KeyFile:                 filepath.Join(boot.InitramfsEncryptionKeyDir, "ubuntu-data.sealed-key"),
-		TPMPolicyUpdateDataFile: filepath.Join(boot.InstallHostFDEDataDir, "policy-update-data"),
-		TPMLockoutAuthFile:      filepath.Join(boot.InstallHostFDEDataDir, "tpm-lockout-auth"),
-	}
-
-	if err := secboot.SealKey(key, &sealKeyParams); err != nil {
-		return fmt.Errorf("cannot seal the encryption key: %v", err)
-	}
+	observer.ChosenEncryptionKey(key)
```

Comment:
this should follow the pattern of the other places in gadget that take observer, and have an if observer != nil